server/crashmanager/views.py

```diff
@@ -1034,6 +1034,35 @@ def partial_update(self, request, pk=None):
         return Response(CrashEntrySerializer(obj).data)
 
 
+class ExternalBugViewSet(mixins.UpdateModelMixin,
+                         viewsets.GenericViewSet):
+    """API endpoint that allows assigning an external Bug to a Bucket"""
+    authentication_classes = (TokenAuthentication, SessionAuthentication)
+    serializer_class = ExternalBugSerializer
+
+    def update(self, request, *args, **kwargs):
+        serializer = self.get_serializer(data=request.data)
+        serializer.is_valid(raise_exception=True)
+
+        entry = get_object_or_404(CrashEntry, id=self.kwargs["crashid"])
+        check_authorized_for_crash_entry(request, entry)
+
+        if not entry.bucket:
+            raise ValidationError("Cannot assign an external bug to an issue that is not associated "
+                                  "to a bucket/signature")
+
+        extBug, _ = Bug.objects.get_or_create(
+            externalId=serializer.validated_data.get('bug_id'),
+            defaults={
+                "externalType": serializer.validated_data.get('provider')
+            }
+        )
+        entry.bucket.bug = extBug
+        entry.bucket.save(update_fields=['bug'])
+
+        return Response(ExternalBugSerializer({'entry': entry, **serializer.validated_data}).data)
```

Comment:
Please add a small comment to explain that you need to send the entry here (and that's why you need to use `update` instead of `perform_update`)